### PR TITLE
Add optional member flag to `prepare`

### PR DESCRIPTION
**Pull request by prestontw**
If specified, and if the top-level cargo.toml
has a members list, replace the list with
only the specified member. 

This is an attempt to address https://github.com/LukeMathWalker/cargo-chef/issues/125. 
If you have feedback on naming or code structure (I feel a little weird placing this function in `mod.rs`),
please let me know! I am happy to make those changes.

I have verified that this works for https://github.com/prestontw/cargo-chef-workspace-experiment

### CODE REVIEW

File: src/skeleton/mod.rs [hunk 1]
```diff
@@ -29,9 +29,15 @@ pub(in crate::skeleton) struct ParsedManifest {
...
     /// Find all Cargo.toml files in `base_path` by traversing sub-directories recursively.
-    pub fn derive<P: AsRef<Path>>(base_path: P) -> Result<Self, anyhow::Error> {
+    pub fn derive<P: AsRef<Path>>(
+        base_path: P,
+        member: Option<String>,
+    ) -> Result<Self, anyhow::Error> {
         // Read relevant files from the filesystem
         let config_file = read::config(&base_path)?;
+        if let Some(member) = member {
+            replace_members_with_specific(&base_path, member)?;
```
Comment (prestontw) [inline src/skeleton/mod.rs:39]: This makes `derive` produce a side effect (modifies the file system). We do this shortly after `derive` (in `main` after preparing the skeleton), but this seemed easier to me than trying to exclude the top-level Cargo.toml if it's both a workspace and `member` is specified.

File: src/main.rs [hunk 2]
```diff
@@ -52,6 +52,13 @@ pub struct Prepare {
     /// It defaults to "recipe.json".
     #[clap(long, default_value = "recipe.json")]
     recipe_path: PathBuf,
+
+    /// Optional: the member in the workspace we wish to compile.
+    /// Note that this is useful in scenarios where we don't copy
+    /// the whole workspace over. Cargo will complain that it can't
+    /// find those members, so this replaces all of the members.
```
Comment (LukeMathWalker) [inline src/main.rs:59]: I don't think we need to mention this here. Alternatively, we should change the language. Something along the lines of:

```
When --bin is specified, `cargo-chef` will ignore all members of the workspace that are not necessary to successfully compile the specific binary.
```

File: src/main.rs [hunk 3]
```diff
@@ -52,6 +52,13 @@ pub struct Prepare {
     /// It defaults to "recipe.json".
     #[clap(long, default_value = "recipe.json")]
     recipe_path: PathBuf,
+
+    /// Optional: the member in the workspace we wish to compile.
```
Comment (LukeMathWalker) [inline src/main.rs:56]:
```suggestion
    /// Optional: the name of the binary we wish to compile.
```

File: src/skeleton/mod.rs (hunk 1 above)
Comment (LukeMathWalker) [reply src/skeleton/mod.rs:39]: I am not a fan - we should manipulate the manifest in memory in `derive` and keep it side-effect free.
You should be able to do this easily by iterating over the `serialised_manifests` collection.

Comment (prestontw) [reply src/skeleton/mod.rs]: That makes sense to me! I'm doing it in `manifests`, though, instead of `serialized_manifests`, to avoid serializing then deserializing.

**Comment by prestontw** [comment]
Hey, @LukeMathWalker! Any other feedback? Is there anything I can do to help get this merged in?

**Comment by LukeMathWalker** [comment]
I am on holiday for the next couple of weeks. I'll try to do a PR review spree when I am back :)

**Comment by prestontw** [comment]
> I am on holiday for the next couple of weeks. I'll try to do a PR review spree when I am back :)

Thank you! Enjoy your time off! :D

File: src/main.rs [hunk 4]
```diff
@@ -136,6 +145,7 @@ fn _main() -> Result<(), anyhow::Error> {
             workspace,
             offline,
             no_std,
+            bin: _bin,
```
Comment (LukeMathWalker) [inline src/main.rs:148]: Shouldn't we pass this over in order to invoke `cargo build` with the `--bin` flag?

Comment (prestontw) [reply src/main.rs]: Added in 438f86a! Thank you for the review!

File: src/recipe.rs [hunk 5]
```diff
@@ -80,7 +81,8 @@ fn build_dependencies(args: &CookArgs) {
         package,
         workspace,
         offline,
-        ..
```
Comment (prestontw) [inline src/recipe.rs:83]: I removed this so the compiler would help direct new contributors to where they should add command functionality if they add a new field, but I'm happy to revert this change if it doesn't match the project idioms.

**Review by LukeMathWalker — APPROVED** [review]

**Comment by LukeMathWalker** [comment]
Thank you for all the work on this!
